pkg/pipeline/DeploymentPipelineConfigService.go

```diff
@@ -386,24 +399,32 @@ func (impl *CdPipelineConfigServiceImpl) CreateCdPipelines(pipelineCreateRequest
 	}
 
 	// TODO: creating git repo for all apps irrespective of acd or helm
-	if isGitOpsConfigured && isGitOpsRequiredForCD {
-
-		gitopsRepoName, chartGitAttr, err := impl.appService.CreateGitopsRepo(app, pipelineCreateRequest.UserId)
-		if err != nil {
-			impl.logger.Errorw("error in creating git repo", "err", err)
-			return nil, err
-		}
-
-		err = impl.RegisterInACD(gitopsRepoName, chartGitAttr, pipelineCreateRequest.UserId, ctx)
+	if isGitOpsConfigured && isGitOpsRequiredForCD && !pipelineCreateRequest.IsCloneAppReq {
+		chart, err := impl.chartRepository.FindLatestChartForAppByAppId(app.Id)
 		if err != nil {
-			impl.logger.Errorw("error in registering app in acd", "err", err)
+			impl.logger.Errorw("Error in fetching latest chart for pipeline", "err", err, "appId", app.Id)
 			return nil, err
 		}
-
-		err = impl.updateGitRepoUrlInCharts(pipelineCreateRequest.AppId, chartGitAttr, pipelineCreateRequest.UserId)
-		if err != nil {
-			impl.logger.Errorw("error in updating git repo url in charts", "err", err)
-			return nil, err
+		if ChartsUtil.IsGitOpsRepoNotConfigured(chart.GitRepoUrl) {
```

Comment:
after using chartService function, implement this method on result struct

Reply:
ChartsUtil.IsGitOpsRepoNotConfigured(chart.GitRepoUrl): is also used at places where we bind the repository struct with JOIN query. There we can not use the above implementation